Description: DANOS commit-9e70806, rebased onto net-snmp 5.9.5.2
Origin: vendor, https://github.com/danos/net-snmp debian/patches

diff --git a/agent/snmpd.c b/agent/snmpd.c
index fe910737b5..4155637286 100644
--- a/agent/snmpd.c
+++ b/agent/snmpd.c
@@ -1024,9 +1024,9 @@ main(int argc, char *argv[])
         if (info) {
             DEBUGMSGTL(("snmpd/main", "Supplementary groups for %s.\n", info->pw_name));
             if (initgroups(info->pw_name, (gid != 0 ? (gid_t)gid : info->pw_gid)) == -1) {
-                snmp_log_perror("initgroups failed");
                 if (!netsnmp_ds_get_boolean(NETSNMP_DS_APPLICATION_ID, 
                                             NETSNMP_DS_AGENT_NO_ROOT_ACCESS)) {
+                    snmp_log_perror("initgroups failed");
                     goto out;
                 }
             }
